packages/foam-language-server/src/index.ts

```diff
@@ -0,0 +1,125 @@
+import {
+	createConnection,
+	ProposedFeatures,
+	InitializeParams,
+	DidChangeConfigurationNotification,
+	CompletionItem,
+	CompletionItemKind,
+	TextDocumentPositionParams,
+	TextDocumentSyncKind,
+	InitializeResult
+} from 'vscode-languageserver';
+
+const data = [
+	{
+		text: 'Foam',
+		displayName: 'Foam',
+		description: 'Foam is a personal knowledge management and sharing system built on Visual Studio Code and GitHub.'
+	},
+	{
+		text: 'Roam',
+		displayName: 'Roam Research',
+		description: 'Roam is a A note-taking tool for networked thought',
+	},
+	{
+		text: 'Obsidian',
+		displayName: "Obsidian",
+		description: 'Obsidian is a powerful knowledge base that works on top of a local folder of plain text Markdown files.'
+	}
+];
+
+// Create a connection for the server, using Node's IPC as a transport.
+// Also include all preview / proposed LSP features.
+let connection = createConnection(ProposedFeatures.all);
+
+let hasConfigurationCapability: boolean = false;
+let hasWorkspaceFolderCapability: boolean = false;
+
+connection.onInitialize((params: InitializeParams) => {
+	let capabilities = params.capabilities;
+
+	// Does the client support the `workspace/configuration` request?
+	// If not, we fall back using global settings.
+	hasConfigurationCapability = !!(
+		capabilities.workspace && !!capabilities.workspace.configuration
+    );
+    
+	hasWorkspaceFolderCapability = !!(
+		capabilities.workspace && !!capabilities.workspace.workspaceFolders
+    );
+
+	const result: InitializeResult = {
+		capabilities: {
+			textDocumentSync: TextDocumentSyncKind.Incremental,
+			// Tell the client that this server supports code completion.
+			completionProvider: {
+				triggerCharacters: ["["],
+				resolveProvider: true
+			}
+		}
+    };
+    
+	if (hasWorkspaceFolderCapability) {
+		result.capabilities.workspace = {
+			workspaceFolders: {
+				supported: true
+			}
+		};
+	}
+	return result;
+});
+
+connection.onInitialized(() => {
+	if (hasConfigurationCapability) {
+		// Register for all configuration changes.
+		connection.client.register(DidChangeConfigurationNotification.type, undefined);
+	}
+	if (hasWorkspaceFolderCapability) {
+		connection.workspace.onDidChangeWorkspaceFolders(_event => {
+			connection.console.log('Workspace folder change event received.');
+		});
+	}
+});
+
+connection.onDidChangeConfiguration(change => {
+	if (hasConfigurationCapability) {
+
```

Comment:
To be implemented. This is boilerplate from lsp-sample, which we'll need once we implement config management.